database/postgres/dal/refresh.py

```diff
@@ -0,0 +1,121 @@
+from sqlalchemy.orm import Session
+from sqlalchemy.exc import SQLAlchemyError
+from typing import Optional
+from datetime import datetime
+
+from database.models.refresh_token_tracking import RefreshTokenTracking
+from database.models.token_blocklist import TokenBlocklist
+from database.postgres.dto.refresh import refreshDTO
+
+class RefreshTokenDAL:
+    def __init__(self, db_session: Session):
+        self.db_session = db_session
+
+    def save_refresh_token(self, refresh_dto: refreshDTO, last_ip: str, last_device: str, nonce: str) -> Optional[int]:
+        try:
+            refresh_entry = None
+            
+            if refresh_dto.id:
+                refresh_entry = self.get_refresh_token_by_id(refresh_dto.id)
+
+            if refresh_entry:
+                refresh_entry.user_id = refresh_dto.user_id
```

Comment:
After u create a DTO, try automapping this fields. You do way too much manual mapping here. I could bet there are libs that'd do the job for ya